packages/backend/src/accounting/psql/ledger-transfer/index.ts

```diff
@@ -67,42 +67,67 @@ export async function getAccountTransfers(
   )
 }
 
-export async function voidTransfer(
+export async function voidTransfers(
   deps: ServiceDependencies,
-  transferRef: string
+  transferRefs: string[]
 ): Promise<void | TransferError> {
-  return updateTransferState(deps, transferRef, LedgerTransferState.VOIDED)
+  return updateTransferState(deps, transferRefs, LedgerTransferState.VOIDED)
 }
 
-export async function postTransfer(
+export async function postTransfers(
   deps: ServiceDependencies,
-  transferRef: string
+  transferRefs: string[]
 ): Promise<void | TransferError> {
-  return updateTransferState(deps, transferRef, LedgerTransferState.POSTED)
+  return updateTransferState(deps, transferRefs, LedgerTransferState.POSTED)
 }
 
 async function updateTransferState(
   deps: ServiceDependencies,
-  transferRef: string,
+  transferRefs: string[],
   state: LedgerTransferState.POSTED | LedgerTransferState.VOIDED
 ): Promise<void | TransferError> {
-  return await deps.knex.transaction(async (trx) => {
-    const transfer = await LedgerTransfer.query(trx)
-      .findOne({ transferRef })
-      .forUpdate()
+  const trx = await deps.knex.transaction()
 
-    if (!transfer) {
-      return TransferError.UnknownTransfer
-    }
+  const updateTransfers = async () => {
+    for (const transferRef of transferRefs) {
+      if (!isValidUuid(transferRef)) {
+        return TransferError.InvalidId
+      }
+
+      const transfer = await LedgerTransfer.query(trx)
+        .findOne({ transferRef })
+        .forUpdate()
+
+      if (!transfer) {
+        return TransferError.UnknownTransfer
+      }
+
+      const error = validateTransferStateUpdate(transfer)
 
-    const transferError = validateTransferStateUpdate(transfer)
+      if (error) {
+        return error
+      }
 
-    if (transferError) {
-      return transferError
+      await transfer.$query(trx).patch({ state })
     }
+  }
+
+  const error = await updateTransfers()
 
-    await transfer.$query(trx).patch({ state })
-  })
+  if (error) {
+    await trx.rollback()
+    return error
+  }
+
+  try {
+    await trx.commit()
```

Comment:
Should `updateTransfers` be inside the `try`/`catch`?

Reply:
would any of the errors actually end up happening in `trx.commit()`?

Reply:
Probably safer anw